components/brave_ads/core/internal/user_engagement/conversions/conversions_new_tab_page_ad_util_unittest.cc

```diff
@@ -0,0 +1,140 @@
+/* Copyright (c) 2024 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at https://mozilla.org/MPL/2.0/. */
+
+#include "base/test/scoped_feature_list.h"
+#include "brave/components/brave_ads/core/internal/ad_units/ad_test_util.h"
+#include "brave/components/brave_ads/core/internal/common/unittest/unittest_base.h"
+#include "brave/components/brave_ads/core/internal/common/unittest/unittest_time_util.h"
+#include "brave/components/brave_ads/core/internal/settings/settings_test_util.h"
+#include "brave/components/brave_ads/core/internal/user_engagement/ad_events/ad_event_builder.h"
+#include "brave/components/brave_ads/core/internal/user_engagement/ad_events/ad_event_info.h"
+#include "brave/components/brave_ads/core/internal/user_engagement/conversions/conversions_util.h"
+#include "brave/components/brave_ads/core/public/ad_units/ad_info.h"
+#include "brave/components/brave_ads/core/public/ads_feature.h"
+
+// npm run test -- brave_unit_tests --filter=BraveAds*
+
+namespace brave_ads {
+
+class BraveAdsConversionsNewTabPageAdUtilTest : public UnitTestBase {};
+
+TEST_F(BraveAdsConversionsNewTabPageAdUtilTest, AllowedToConvertViewedAdEvent) {
+  // Arrange
+  const AdInfo ad =
+      test::BuildAd(AdType::kNewTabPageAd, /*should_generate_random_uuids=*/
+                    true);
+  const AdEventInfo ad_event = BuildAdEvent(
+      ad, ConfirmationType::kViewedImpression, /*created_at=*/Now());
+
+  // Act & Assert
+  EXPECT_TRUE(IsAllowedToConvertAdEvent(ad_event));
```

Comment:
This test is passing because by default in tests users are part of Brave Rewards.
```
      // Only if:
      // - The user has opted into new tab page ads and has either joined Brave
      //   Rewards or new tab page ad events should always be triggered.
      return UserHasOptedInToNewTabPageAds() &&
             (UserHasJoinedBraveRewards() ||
              ShouldAlwaysTriggerNewTabPageAdEvents());
```

I have added an additional test to cover the above use case, `NotAllowedToConvertViewedAdEventForNonRewardsUserIfShouldNotAlwaysTriggerBraveNewTabPageAdEvents`.

Thanks :-)